By default the Lua makefile will only create a static library (liblua.a).
This patch will modify the makefile so it will create a dynamic library.
--- Makefile	2017-04-26 18:12:27.442079208 +0200
+++ Makefile	2017-04-26 20:32:34.818881639 +0200
@@ -42,6 +42,9 @@
 TO_BIN= lua luac
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h lua.hpp
 TO_LIB= liblua.a
+TO_LIBX=liblua.so
+TO_LIBXV=liblua.so.$(V)
+TO_LIBXR=liblua.so.$(R)
 TO_MAN= lua.1 luac.1
 
 # Lua version and release.
@@ -52,7 +55,7 @@
 all:	$(PLAT)
 
 $(PLATS) clean:
-	cd src && $(MAKE) $@
+	cd src && $(MAKE) $@ V=$(V) R=$(R)
 
 test:	dummy
 	src/lua -v
@@ -62,12 +65,15 @@
 	cd src && $(INSTALL_EXEC) $(TO_BIN) $(INSTALL_BIN)
 	cd src && $(INSTALL_DATA) $(TO_INC) $(INSTALL_INC)
 	cd src && $(INSTALL_DATA) $(TO_LIB) $(INSTALL_LIB)
+	cd src && $(INSTALL_EXEC) $(TO_LIBX) $(INSTALL_LIB)
+	cd $(INSTALL_LIB) && ln -s $(TO_LIBX) $(TO_LIBXV) && ln -s $(TO_LIBX) $(TO_LIBXR)
 	cd doc && $(INSTALL_DATA) $(TO_MAN) $(INSTALL_MAN)
 
 uninstall:
 	cd src && cd $(INSTALL_BIN) && $(RM) $(TO_BIN)
 	cd src && cd $(INSTALL_INC) && $(RM) $(TO_INC)
 	cd src && cd $(INSTALL_LIB) && $(RM) $(TO_LIB)
+	cd src && cd $(INSTALL_LIB) && $(RM) $(TO_LIBX)
 	cd doc && cd $(INSTALL_MAN) && $(RM) $(TO_MAN)
 
 local:
@@ -90,6 +96,7 @@
 	@echo "TO_BIN= $(TO_BIN)"
 	@echo "TO_INC= $(TO_INC)"
 	@echo "TO_LIB= $(TO_LIB)"
+	@echo "TO_LIBX= $(TO_LIBX)"
 	@echo "TO_MAN= $(TO_MAN)"
 	@echo "INSTALL_TOP= $(INSTALL_TOP)"
 	@echo "INSTALL_BIN= $(INSTALL_BIN)"
